### PR TITLE
Update to some Custom Analyzers API model classes

### PR DESCRIPTION
**Pull request by Yahnoosh**
Marking EdgeNGramTokenFilter, NGramTokenFilter, KeywordTokenizer and StandardTokenizer as obsolete and adding V2 versions for all of them.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@Yahnoosh__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (jlembicz). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by brjohnstmsft** [comment]
@Yahnoosh Please prefix all SDK commits and pull requests with "Search SDK: "

File: src/Search/Search.Tests/Tests/CustomAnalyzerTests.cs [hunk 1]
```diff
@@ -650,7 +650,7 @@ private static void AssertIndexContainsAllAnalysisComponents(params Index[] inde
                 select new { Type = typeGroup.Key, Count = typeGroup.Count() };
 
             Dictionary<Type, int> instanceCountMap = analysisTypeCounts.ToDictionary(tc => tc.Type, tc => tc.Count);
-            IEnumerable<Type> allAnalysisComponentTypes = typeof(Index).GetTypeInfo().Assembly.ExportedTypes.Where(IsAnalysisComponentType);
+            IEnumerable<Type> allAnalysisComponentTypes = typeof(Index).GetTypeInfo().Assembly.ExportedTypes.Where((Type type) => IsAnalysisComponentType(type) && !IsDeprecatedType(type));
```
Comment (brjohnstmsft) [inline src/Search/Search.Tests/Tests/CustomAnalyzerTests.cs]: You don't need to type lambda parameters, and if there's only one, you don't need parentheses either.

**Review by bitsforthought — CHANGES_REQUESTED** [review]
I left some comments on the corresponding Swagger review. You'll need to regenerate the code and maybe re-apply some tweaks. See the Swagger review for details.

**Comment by Yahnoosh** [comment]
@brjohnstmsft thanks, I'll update this PR once I get a sign of on the Swagger changes. A new CR is out.

**Pull request closed without merging**